src/node.cc

```diff
@@ -2741,6 +2798,16 @@ void SetupProcessObject(Environment* env,
   process->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "execArgv"),
                exec_arguments);
 
+  // process.allowedEnvironmentNodeFlags
+  Local<Array> env_flags = Array::New(env->isolate(),
+                                      arraysize(environment_flags));
+  for (unsigned int i = 0; i < arraysize(environment_flags); ++i) {
+    env_flags->Set(i, String::NewFromUtf8(env->isolate(),
+        environment_flags[i]));
+  }
+  process->Set(FIXED_ONE_BYTE_STRING(env->isolate(),
+                                     "allowedEnvironmentNodeFlags"), env_flags);
+
```

Comment:
I think the array should be frozen much like [`Module.builtinModules`](https://nodejs.org/dist/latest-v10.x/docs/api/all.html#modules_module_builtinmodules). The initial array can be created in C++ but in the bootstrap somewhere it can be augmented to be frozen if possible.

Reply:
seems fair to me; is there a precedent or example elsewhere?

Reply:
There is on the JS side. See https://github.com/nodejs/node/commit/c5093fceb56ddc3126a7a9c4807a64fb10296b9d. The array could be frozen somewhere in https://github.com/nodejs/node/blob/master/lib/internal/bootstrap/node.js.

Reply:
this is done (see [here](https://github.com/boneskull/node/commit/a95529a32e2c4adb3981562a3ca01ea85959f171#diff-362950ada44657730d806a6908957c35R192)), but there isn't any obvious place for it to live.  if someone has a better idea, please suggest!